rqt_py_common/src/rqt_py_common/topic_helpers.py

```diff
@@ -28,161 +28,243 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-import roslib
-import roslib.msgs
-import roslib.message
-from rostopic import get_topic_type
-from python_qt_binding.QtCore import qDebug
+import rclpy
+import math
+
+from rclpy import logging
+from ament_index_python import get_resource
+
+
+def get_topic_names_and_types():
+    shutdown_rclpy = False
+    if not rclpy.ok():
+        shutdown_rclpy = True
+        rclpy.init()
+
+    node = rclpy.create_node("TopicHelpers__get_topic_names_and_types")
+    # Give the node time to learn about the graph
+    rclpy.spin_once(node, timeout_sec=0.05)
+
+    topic_list = node.get_topic_names_and_types()
+
+    node.destroy_node()
+    if shutdown_rclpy:
+        rclpy.shutdown()
+
+    return topic_list
+
+
+_logger = logging.get_logger("topic_helpers")
```

Comment:
Maybe instead of creating a global variable, you just use
logging.get_logger().info()

Reply:
addressed